### PR TITLE
Feature/delete product and its review

### PR DESCRIPTION
**Pull request by amaan5555**
Done the backend part

### CODE REVIEW

**Comment by kd-labs** [comment]
@amaan5555 Please change the target branch to env/sbox

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "ejs": "^3.1.9",
     "ejs-mate": "^4.0.0",
     "express": "^4.18.2",
-    "mongoose": "^7.3.4"
+    "mongoose": "^7.4.1"
   },
```
Comment (sabeelkhan99) [inline package.json]: Why are updating mongoose version here? Any specific reason if no please revert it to 7.3.4, you can use the command `npm i mongoose@7.3.4`

File: src/repositories/productRepo.js [hunk 2]
```diff
@@ -9,12 +10,23 @@ const save = (product) => {
 
 const findById = (id) => Product.findById(id);
 
+
+const deleteProductAndReviews = (productId) => {
+  
+       Product.findByIdAndDelete(productId);
+  
+       Review.deleteMany({ productId: productId });
```
Comment (sabeelkhan99) [inline src/repositories/productRepo.js:18]: 1. Do we have a `productId` in our reviews?
2. Is this working on your local?
3. And why are we not waiting for the promise to get resolved.
4. Check and try running on your local there seems to a wrong logic being applied here.

File: src/services/productService.js [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
+const Review = require('../models/reviews');
 
 const getAllProducts = async () => productRepo.getAllProducts();
 
 const create = async (product) => productRepo.save(product);
 
 const findById = async (id) => productRepo.findByIdWithReviews(id);
 
+const deleteProductAndReviews = async (productId) => productRepo.deleteProductAndReviews(productId);
+
```
Comment (sabeelkhan99) [inline src/services/productService.js:15]: Why are we not returing promise from here? How are you ensuring the product is being deleted here?

**Pull request closed without merging**